python/ray/tune/experiment/trial.py

```diff
@@ -186,6 +191,58 @@ def _create_unique_logdir_name(root: str, relative_logdir: str) -> str:
     return relative_logdir
 
 
+def _noop_logger_creator(
+    config: Dict[str, Any], logdir: str, should_chdir: bool = True
+):
+    # Upon remote process setup, record the actor's original working dir before
+    # changing to the Tune logdir
+    os.environ.setdefault("TUNE_ORIG_WORKING_DIR", os.getcwd())
+
+    os.makedirs(logdir, exist_ok=True)
+    if should_chdir:
+        # Set the working dir to the trial directory in the remote process,
+        # for user file writes
+        if not ray._private.worker._mode() == ray._private.worker.LOCAL_MODE:
+            os.chdir(logdir)
+    return NoopLogger(config, logdir)
```

Comment:
Note for future: We should remove this `NoopLogger` stuff. Looks like the old `Logger` interface is only possible if you directly create Trainable directly. The regular case will always use this `NoopLogger`.

Btw, what was the reason for switching to the `LoggerCallback`?

Reply:
Agreed, we should remove it. Note that it's currently referred to in rllib, and might still be relevant for RLLib algorithms that run without tune.